capstone-backend/app.js

```diff
@@ -315,36 +315,62 @@ const retrieveMatchData = async (limit, offset, currentUser) => {
     return ({ matchesInfo: matchesInfo, matchResults: matchResults, matchMessage: "Matches Retrieved!", typeStatus: "success" });
 }
 
-app.post('/login', async (req, res) => {
-    Parse.User.enableUnsafeCurrentUser();
+app.post('/reset-session', async (req, res) => {
     const infoUser = req.body;
 
     try {
-        const user = await Parse.User.logIn(infoUser.email, infoUser.password);
+        const user = await Parse.User.become(infoUser.sessionToken);
         const rawdata = fs.readFileSync('data/majors.json');
         const majors = JSON.parse(rawdata);
         res.send({ userInfo: user, loginMessage: "User logged in!", typeStatus: "success", infoUser: infoUser, majors: majors });
     } catch (error) {
+        console.log("error", error)
+        res.send({ loginMessage: error, typeStatus: "danger", infoUser: infoUser });
+    }
+})
+
+app.post('/login', async (req, res) => {
+    const infoUser = req.body;
+    try {
+        const user = await Parse.User.logIn(infoUser.email, infoUser.password);
+        const sessionToken = user.getSessionToken();
+        const rawdata = fs.readFileSync('data/majors.json');
+        const majors = JSON.parse(rawdata);
+        res.send({ userInfo: user, sessionToken: sessionToken, loginMessage: "User logged in!", typeStatus: "success", infoUser: infoUser, majors: majors });
+    } catch (error) {
+        console.log("error", error)
         res.send({ loginMessage: error, typeStatus: "danger", infoUser: infoUser });
     }
 })
 
 app.post('/logout', async (req, res) => {
 
     try {
-        await Parse.User.logOut();
-        res.send({ logoutMessage: "User logged out!", typeStatus: "success" });
+        let query = new Parse.Query("_Session")
+        query.equalTo("sessionToken", req.body.sessionToken)
```

Comment:
what does this do? does it return a bool? and we are not using that value?

Reply:
The query contains the current user associated with the session token. The following lines (await query.first({ useMasterKey: true }).then(function (user){.....}) log out the current user returned by the query.